substrate/pkg/controller/substrate/cluster/addons/rbac.go

```diff
@@ -0,0 +1,54 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package addons
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/awslabs/kit/substrate/pkg/apis/v1alpha1"
+	nodebootstraptokenphase "k8s.io/kubernetes/cmd/kubeadm/app/phases/bootstraptoken/node"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type RBAC struct {
+}
+
+func (r *RBAC) Create(ctx context.Context, substrate *v1alpha1.Substrate) (reconcile.Result, error) {
+	client, err := KubeClientFor(ctx, substrate)
+	if errors.Is(err, ErrWaitingForSubstrateEndpoint) {
+		return reconcile.Result{Requeue: true}, nil
+	} else if err != nil {
+		return reconcile.Result{}, err
+	}
+	// Create RBAC rules that makes the bootstrap tokens able to post CSRs
+	if err := nodebootstraptokenphase.AllowBootstrapTokensToPostCSRs(client); err != nil {
```

Comment:
Seems like you're somewhat forced into this pattern by kubeadm. Do you know how long each of these calls take?

Reply:
Its less than a second.